test/test_stokes.py

```diff
@@ -311,6 +320,259 @@ def test_exterior_stokes_2d(ctx_factory, qbx_order=3):
+            rhs,
+            x0=rhs,
+            tol=gmres_tol,
+            progress=True,
+            stall_iterations=0,
+            hard_failure=True)
+
+    sigma = result.solution
+
+    # }}}
+
+    # {{{ check velocity on surface
+
+    def rnorm2(x, y):
+        y_norm = actx.np.linalg.norm(y.dot(y), ord=2)
+        if y_norm < 1.0e-14:
+            y_norm = 1.0
+
+        d = x - y
+        return actx.np.linalg.norm(d.dot(d), ord=2) / y_norm
+
+    h_max = bind(places, sym.h_max(ambient_dim))(actx)
+
+    if ambient_dim == 3:
+        sym_surf_velocity = op.velocity(sym_sigma,
+                normal=sym_normal, mu=sym_mu,
+                qbx_forced_limit="avg")
+        sym_surf_velocity = -0.5 * op.side * sym_sigma + sym_surf_velocity
+
+        surf_velocity = bind(places, sym_surf_velocity)(actx,
+                sigma=sigma, **op_context)
+
+        s_error = rnorm2(surf_velocity, bc)
+        logger.info("resolution %4d h_max %.5e error %.5e",
+                resolution, h_max, s_error)
+        assert s_error < gmres_tol
+
+    # }}}
+
+    # {{{ check velocity at "point_target"
+
+    ps_velocity = bind(places, sym_velocity,
+            auto_where=("source", "point_target"))(actx, sigma=sigma, **op_context)
+    ex_velocity = bind(places, sym_source_pot,
+            auto_where=("point_source", "point_target"))(actx, sigma=charges, mu=mu)
+
+    v_error = rnorm2(ps_velocity, ex_velocity)
+
+    logger.info("resolution %4d h_max %.5e error %.5e",
+            resolution, h_max, v_error)
+
+    # }}}}
+
+    # {{{ visualize
+
+    if not visualize:
+        return h_max, v_error
+
+    from meshmode.discretization.visualization import make_visualizer
+    vis = make_visualizer(actx, density_discr, target_order)
+
+    filename = "stokes_solution_{}d_{}_ovsmp_{}.vtu".format(
+            ambient_dim, resolution, source_ovsmp)
+
+    vis.write_vtk_file(filename, [
+        ("density", sigma),
+        ("bc", bc),
+        ("rhs", rhs),
+        ], overwrite=True)
+
+    # }}}
+
+    return h_max, v_error
+
+
+@pytest.mark.slowtest
+@pytest.mark.parametrize("ambient_dim", [2, 3])
+def test_exterior_stokes(ctx_factory, ambient_dim, visualize=False):
+    from pytools.convergence import EOCRecorder
+    eoc = EOCRecorder()
+
+    target_order = 3
+    qbx_order = 3
+
+    if ambient_dim == 2:
+        resolutions = [20, 35, 50]
+    elif ambient_dim == 3:
+        resolutions = [0, 1, 2]
+    else:
+        raise ValueError(f"unsupported dimension: {ambient_dim}")
+
+    for resolution in resolutions:
+        h_max, err = run_exterior_stokes(ctx_factory,
+                ambient_dim=ambient_dim,
+                target_order=target_order,
+                qbx_order=qbx_order,
+                resolution=resolution,
+                visualize=visualize)
+
+        eoc.add_data_point(h_max, err)
+
+    print(eoc)
+    assert eoc.order_estimate() > target_order - 0.5
```

Comment:
```suggestion
    # This convergence data is not as clean as it could be. See
    # https://github.com/inducer/pytential/pull/32
    # for some discussion.
    assert eoc.order_estimate() > target_order - 0.5
```